ops/model.py

```diff
@@ -758,3 +760,55 @@ def network_get(self, binding_name, relation_id=None):
             if 'relation not found' in str(e):
                 raise RelationNotFoundError() from e
             raise
+
+    def add_metrics(self, metrics, labels=None):
+        cmd = ['add-metric']
+
+        if labels:
+            label_args = []
+            for k, v in labels.items():
+                _ModelBackendValidator.validate_metric_label(k)
+                _ModelBackendValidator.validate_label_value(v)
+                label_args.append(f'{k}={v}')
+            cmd.extend(['--labels', ','.join(label_args)])
+
+        metric_args = []
+        for k, v in metrics.items():
+            _ModelBackendValidator.validate_metric_key(k)
+            metric_value = _ModelBackendValidator.format_metric_value(v)
+            metric_args.append(f'{k}={metric_value}')
+        cmd.extend(metric_args)
+        self._run(*cmd)
+
+
+class _ModelBackendValidator:
```

Comment:
I like how the code looks using this, so I stand by my previous suggestions. Thank you for implementing them!

I suspect the name could be better, but I don't have anything to offer on that front, sorry.